backend/src/test/java/com/htecgroup/skynest/service/impl/FolderServiceImplTest.java

```diff
@@ -85,6 +90,35 @@ void getFolderDetails() {
     verify(folderRepository, times(1)).findById(any());
   }
 
+  @Test
+  void editFolder() {
+    FolderEntity expectedFolderEntity = FolderEntityUtil.getFolderWithoutParent();
+    when(folderRepository.findById(any())).thenReturn(Optional.of(expectedFolderEntity));
+    when(folderRepository.save(any())).thenReturn(expectedFolderEntity);
+
+    FolderEditRequest folderEditRequest = FolderEditRequestUtil.get();
+    FolderResponse actualFolderResponse =
+        folderService.editFolder(folderEditRequest, expectedFolderEntity.getId());
+
+    this.assertFolderEntityAndFolderResponse(expectedFolderEntity, actualFolderResponse);
+    verify(folderRepository, times(1)).findById(any());
+    verify(folderRepository, times(1)).save(any());
```

Comment:
But he verify that the methods are called just once. It is useful if method is changed so somehow some methods are called more times than needed maybe and then it is recorded here and test will fail.

Reply:
Well I'm not sure, but I think you can either use when() to mock them, or use verify() to verify them. I even think that if it gets called twice and we have it mocked in when() once, it will fail. My point is I think it's redundant to use when() and verify() on the same method.

Reply:
@petarstoimenovskihtec we need guidance

Reply:
Definitely clearer. My mind was super focused on this kind of a situation that i totally forgot about any()..
```
when(folderRepository.findById(id1)).thenReturn(MockObject1);
when(folderRepository.findById(id2)).thenReturn(MockObject2);
```

Reply:
Well yeah, it is always good to verify anything, makes more sense if you are doing it with expected value instead of any(). When using when() you can use any() both with verify() I strongly suggest to use the expected value. Also it is always useful to verify that there aren't any additional interactions with that mock.
` verifyNoMoreInteractions(mockServiceHere);`